crates/subspace-fake-runtime-api/src/lib.rs

```diff
@@ -0,0 +1,481 @@
+            unreachable!()
+        }
+
+        fn initialize_block(_header: &<Block as BlockT>::Header) -> ExtrinsicInclusionMode {
+            unreachable!()
+        }
+    }
+
+    impl sp_api::Metadata<Block> for Runtime {
+        fn metadata() -> OpaqueMetadata {
+            unreachable!()
+        }
+
+        fn metadata_at_version(_version: u32) -> Option<OpaqueMetadata> {
+            unreachable!()
+        }
+
+        fn metadata_versions() -> Vec<u32> {
+            unreachable!()
+        }
+    }
+
+    impl sp_block_builder::BlockBuilder<Block> for Runtime {
+        fn apply_extrinsic(_extrinsic: <Block as BlockT>::Extrinsic) -> ApplyExtrinsicResult {
+            unreachable!()
+        }
+
+        fn finalize_block() -> <Block as BlockT>::Header {
+            unreachable!()
+        }
+
+        fn inherent_extrinsics(_data: sp_inherents::InherentData) -> Vec<<Block as BlockT>::Extrinsic> {
+            unreachable!()
+        }
+
+        fn check_inherents(
+            _block: Block,
+            _data: sp_inherents::InherentData,
+        ) -> sp_inherents::CheckInherentsResult {
+            unreachable!()
+        }
+    }
+
+    impl sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block> for Runtime {
+        fn validate_transaction(
+            _source: TransactionSource,
+            _tx: <Block as BlockT>::Extrinsic,
+            _block_hash: <Block as BlockT>::Hash,
+        ) -> TransactionValidity {
+            unreachable!()
+        }
+    }
+
+    impl sp_offchain::OffchainWorkerApi<Block> for Runtime {
+        fn offchain_worker(_header: &<Block as BlockT>::Header) {
+            unreachable!()
+        }
+    }
+
+    impl sp_objects::ObjectsApi<Block> for Runtime {
+        fn extract_block_object_mapping(_block: Block, _successful_calls: Vec<Hash>) -> BlockObjectMapping {
+            unreachable!()
+        }
+
+        fn validated_object_call_hashes() -> Vec<Hash> {
+            // No pallets produce objects right now
+            unreachable!()
+        }
+    }
+
+    impl sp_consensus_subspace::SubspaceApi<Block, FarmerPublicKey> for Runtime {
+        fn pot_parameters() -> PotParameters {
+            unreachable!()
+        }
+
+        fn solution_ranges() -> SolutionRanges {
+            unreachable!()
+        }
+
+        fn submit_report_equivocation_extrinsic(
+            _equivocation_proof: EquivocationProof<<Block as BlockT>::Header>,
+        ) -> Option<()> {
+            unreachable!()
+        }
+
+        fn submit_vote_extrinsic(
+            _signed_vote: SignedVote<NumberFor<Block>, <Block as BlockT>::Hash, FarmerPublicKey>,
+        ) {
+            unreachable!()
+        }
+
+        fn is_in_block_list(_farmer_public_key: &FarmerPublicKey) -> bool {
+            // TODO: Either check tx pool too for pending equivocations or replace equivocation
+            //  mechanism with an alternative one, so that blocking happens faster
```

Comment:
TODO is not relevant here, same about comment above in `validated_object_call_hashes`